modules/cfe_testcase/src/msg_api_test.c

```diff
@@ -0,0 +1,231 @@
+    UtAssert_True(CFE_MSG_GetNextSequenceCount(0x3FFF) == 0, "CFE_MSG_GetNextSequenceCount(0x3FFF) = 0");
+    UtAssert_True(CFE_MSG_GetNextSequenceCount(2u) == 3u, "CFE_MSG_GetNextSequenceCount(2u) = 3u");
+
+    /* test get/set-eds-version */
+    UtAssert_INT32_EQ(CFE_MSG_GetEDSVersion(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetEDSVersion(&cmd.Msg, 5u));
+    UtAssert_SUCCESS(CFE_MSG_GetEDSVersion(&cmd.Msg, &edsVersion));
+
+    UtAssert_True(edsVersion == 5u, "CFE_MSG_SetEDSVersion => edsVersion = 5u");
+
+    /* test get/set-endian */
+    UtAssert_INT32_EQ(CFE_MSG_SetEndian(&cmd.Msg, CFE_MSG_Endian_Invalid), CFE_MSG_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_MSG_GetEndian(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetEndian(&cmd.Msg, CFE_MSG_Endian_Little));
+    UtAssert_SUCCESS(CFE_MSG_GetEndian(&cmd.Msg, &endian));
+
+    UtAssert_True(endian == CFE_MSG_Endian_Little, "CFE_MSG_SetEndian => endian = CFE_MSG_Endian_Little");
+
+    /* test get/set-playback-flag */
+    UtAssert_INT32_EQ(CFE_MSG_SetPlaybackFlag(&cmd.Msg, CFE_MSG_PlayFlag_Invalid), CFE_MSG_BAD_ARGUMENT);
+    UtAssert_INT32_EQ(CFE_MSG_GetPlaybackFlag(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetPlaybackFlag(&cmd.Msg, CFE_MSG_PlayFlag_Playback));
+    UtAssert_SUCCESS(CFE_MSG_GetPlaybackFlag(&cmd.Msg, &playbackFlag));
+
+    UtAssert_True(playbackFlag == CFE_MSG_PlayFlag_Playback,
+                  "CFE_MSG_SetPlaybackFlag => playbackFlag = CFE_MSG_PlayFlag_Playback");
+
+    /* test get/set-subsystem */
+    UtAssert_INT32_EQ(CFE_MSG_GetSubsystem(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetSubsystem(&cmd.Msg, 3u));
+    UtAssert_SUCCESS(CFE_MSG_GetSubsystem(&cmd.Msg, &subSystem));
+
+    UtAssert_True(subSystem == 3u, "CFE_MSG_SetSubsystem => subSystem = 3u");
+
+    /* test get/set-system */
+    UtAssert_INT32_EQ(CFE_MSG_GetSystem(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetSystem(&cmd.Msg, 3u));
+    UtAssert_SUCCESS(CFE_MSG_GetSystem(&cmd.Msg, &system));
+
+    UtAssert_True(system == 3u, "CFE_MSG_SetSystem => system = 3u");
+
+    /* test generate-checksum */
+    UtAssert_INT32_EQ(CFE_MSG_GenerateChecksum(NULL), CFE_MSG_BAD_ARGUMENT);
+    CFE_MSG_SetType(&cmd.Msg, CFE_MSG_Type_Tlm);
+    UtAssert_INT32_EQ(CFE_MSG_GenerateChecksum(&cmd.Msg), CFE_MSG_BAD_ARGUMENT);
+    CFE_MSG_SetType(&cmd.Msg, CFE_MSG_Type_Cmd);
+
+    /* create new cmd message - TODO */
+    /*
+     * CFE_MSG_CommandHeader_t cmd2;
+     * memset(&cmd2, 0xFF, sizeof(cmd2));
+     * UtAssert_SUCCESS(CFE_MSG_Init(&cmd2.Msg, CFE_SB_ValueToMsgId(1u), sizeof(cmd2));
+     *
+     * UtAssert_UINT32_EQ(CFE_MSG_GenerateChecksum(&cmd2.Msg), <precalculated checksum for a new message>);
+     */
+
+    /*
+    * UtAssert_SUCCESS(CFE_MSG_ValidateChecksum(&cmd.Msg, &isValid));
+    * UtAssert_True(isValid, "checksum: isValid = true");
```

Comment:
Suggested flow - memset cmd to 0, validate checksum (should return isValid of false), generate the checksum (actually fills it in correctly) then validate again and should return isValid of true.